app/services/GuidanceService.scala

```diff
@@ -42,12 +42,20 @@ class GuidanceService @Inject() (
 
   def getProcessContext(sessionId: String): Future[RequestOutcome[ProcessContext]] = sessionRepository.get(sessionId)
 
-  def getProcessContext(sessionId: String, pageUrl: String, previousPageByLink: Boolean): Future[RequestOutcome[ProcessContext]] =
-    sessionRepository.get(sessionId, pageUrl, previousPageByLink)
+  def getProcessContext(sessionId: String, processCode: String, url: String, previousPageByLink: Boolean)
+                       (implicit context: ExecutionContext): Future[RequestOutcome[ProcessContext]] = {
+    val optionalPageHistoryUrl: Option[String] = if (isAuthenticationUrl(url)) None else Some(s"$processCode$url")
+    sessionRepository.get(sessionId, optionalPageHistoryUrl, previousPageByLink).map{ ctx =>
+      (ctx, optionalPageHistoryUrl) match {
+        case (Right(ctx), Some(_)) if !ctx.secure => Left(AuthenticationError)
+        case (result, _) => result
+      }
+    }
+  }
 
   def getPageEvaluationContext(processCode: String, url: String, previousPageByLink: Boolean, sessionId: String)
```

Comment:
Error handling could log failed authentication before returning error to controller

Reply:
Log added to controller before redirect